send-message.js

```diff
@@ -102,8 +102,57 @@ async function sendLatestNews(req, res) {
     });
 }
 
+async function sendHomepage(req, res) {
+  var who_number = req.query.number;
+  if (who_number === undefined) {
+    who_number = "41798931892";
+  }
+  const token = TOKENS[who_number];
+  const client = axios.create({
+    baseURL: TURN_URL,
+    timeout: 300,
+    headers: { Authorization: `Bearer ${token}` }
+  });
+
+  const user = req.body.contacts[0].wa_id;
+  const messageId = req.body.messages[0].id;
+  debug(`/homepage called for ${user} with message id ${messageId}`);
+  const [number, countryCode] = phone(`+${user}`);
+  if (!number) {
+    return res.json({ country: "unknown" });
+  }
+  debug(`The country code is: ${countryCode}`);
+
+  const countryData = await retrieveCountryData(countryCode);
+  const newsList = await retrieveLatestNews();
+
+  const [msg1, msg2] = formatHomepageMessages(countryData, newsList);
+  inspect("homepage stats message:")(msg1);
+  inspect("homepage news message:")(msg2);
+
+  sendMessage(client, messageId, msg1, user)
+    .then(inspect("message response:"))
+    .catch(err => {
+      if (err.response) {
+        inspect("error data")(err.response.data);
+        inspect("error status")(err.response.status);
+        inspect("error headers")(err.response.headers);
+      }
+    });
+  return sendMessage(client, messageId, msg2, user)
+    .then(inspect("message response:"))
+    .catch(err => {
+      if (err.response) {
+        inspect("error data")(err.response.data);
+        inspect("error status")(err.response.status);
+        inspect("error headers")(err.response.headers);
+      }
+    });
```

Comment:
We've got 2 `sendMessage` promises here, so I think the returning promise should involve both in some way.

If we want to chain them (make one request, then make another), then we can use the `.then` syntax to nest them.

If we just want to make 2 requests in parallel (probably what we want, since request order doesn't guarantee message order anyway), then we can do a `return Promise.allSettled([msg1, msg2])`, where `msg1` and `msg2` are the two promises from the `sendMessage` function.

Reply:
Thanks @rudigiesler, I decided to go with the chaining option. We do actually care about the message order but managing that and getting the second message to depend on the delivery of the first one is something we're looking into later.